### PR TITLE
Support authorization via service_token/service_id

### PR DESCRIPTION
**Pull request by tigrankhachikyan**
I've made a minor changes to support authorization with service_token/service_id.

* The reason I made these changes was, that I have some trouble when trying to authorize with only provider_key (provider key is invalid).
This change is not a solution for invalid provider key. Instead it is just an enhancement.

### CODE REVIEW

File: lib/Net/ThreeScale/Client.pm [hunk 1]
```diff
@@ -41,14 +41,16 @@ BEGIN {
 sub new {
 	my $class = shift;
 	my $params = ( $#_ == 0 ) ? { %{ (shift) } } : {@_};
-
 	my $agent_string = $params->{user_agent} || $DEFAULT_USER_AGENT;
 
-	croak("provider_key is a required parameter")
-		unless $params->{provider_key};
+	croak("provider_key or service_token/service_id pair are required")
+		unless $params->{provider_key} || ( $params->{service_token} && $params->{service_id});
```
Comment (slobo) [inline lib/Net/ThreeScale/Client.pm]: probably `xor` would make sense here since it's either one or the other

File: lib/Net/ThreeScale/Client.pm [hunk 2]
```diff
@@ -153,7 +159,9 @@ sub report {
 		unless (is_arrayref($p->{transactions}));
 
 	my %query = (
-		provider_key => $self->{provider_key},
+    (provider_key  => $self->{provider_key})x!!  $self->{provider_key},
```
Comment (slobo) [inline lib/Net/ThreeScale/Client.pm]: i think original code is tabs instead of spaces

File: lib/Net/ThreeScale/Client.pm [hunk 3]
```diff
@@ -123,7 +127,9 @@ sub authrep {
 	die("user_key is required") unless defined($p->{user_key});
 
 	my %query = (
-		provider_key => $self->{provider_key},
+    (provider_key  => $self->{provider_key})x!!  $self->{provider_key},
```
Comment (slobo) [inline lib/Net/ThreeScale/Client.pm]: same here, should be tabs

**Comment by slobo** [comment]
It seems that since November 2016, `service_id` became a mandatory parameter. 

So perhaps `service_token` handling should be decoupled from it - regardless of using `provider_key` or `service_token`, `service_id` is still needed.

**Comment by davel** [comment]
I have fixed the formatting and merged these changes.

@tigrankhachikyan Do you have an opinion on https://github.com/3scale/3scale_ws_api_for_perl/pull/14#issuecomment-274964059 ?

**Pull request closed without merging**